### ISSUE TITLE
score method of KerasRegressor class returns a loss value which causes BaseSearchCV class to select the worst model

### ISSUE THREAD

**Issue by avielbl** [fchollet/keras#5577]
Hi,

I found that by using the KerasRegressor wrapper for use with sklearn RandomizedSearchCV or GridSearchCV the worst model is returned by means of highest validation loss.

I believe it is resulted from the fact the score method returns a loss (best model is with lowest value) as the score while scikit expect a score value such as validation accuracy (best model is with highest value).

I temporarily fixed that by returning -loss:

    def score(self, X, y, **kwargs):
        '''Returns the mean loss on the given test data and labels.

        # Arguments
            X: array-like, shape `(n_samples, n_features)`
                Test samples where n_samples in the number of samples
                and n_features is the number of features.
            y: array-like, shape `(n_samples,)`
                True labels for X.
            kwargs: dictionary arguments
                Legal arguments are the arguments of `Sequential.evaluate`.

        # Returns
            score: float
                Mean accuracy of predictions on X wrt. y.
        '''
        kwargs = self.filter_sk_params(Sequential.evaluate, kwargs)
        loss = self.model.evaluate(X, y, **kwargs)
        if type(loss) is list:
            return -loss[0]
        return -loss

**Comment by pauleikis** [comment]
Came here to report the same bug and I have arrived at exactly same conclusions, that the return value of the score method should be multiplied by -1

**Comment by 10wei1000** [comment]
I am also shuffling from this bug, please fix it. Many thanks!

**Comment by paolof89** [comment]
Hi,
I found the same bug, I used the scoring parameter of the GridSearchCV function to solve the issue.
I used the @avielbl code to create a custom function, so I don't need to modify the KerasRegressor implementation.

`def my_score(estimator, X, y, **kwargs):
    from keras.models import Sequential
    kwargs = estimator.filter_sk_params(Sequential.evaluate, kwargs)
    loss = estimator.model.evaluate(X, y, **kwargs)
    if type(loss) is list:
        return -loss[0]
    return -loss
grid = GridSearchCV(estimator=model, param_grid=param_grid, scoring=my_score)`
